src/Psalm/Config.php

```diff
@@ -1013,8 +1013,16 @@ private static function fromXmlAndPaths(
             chdir($config->base_dir);
         }
 
-        if (is_dir($config->cache_directory) === false && @mkdir($config->cache_directory, 0777, true) === false) {
-            error_log('Could not create cache directory: ' . $config->cache_directory);
+        if (!is_dir($config->cache_directory)) {
+            try {
+                mkdir($config->cache_directory, 0777, true);
```

Comment:
seems weird to not check mkdir return. Does it throws on any error with some config?